x/data/server/hasher/hasher.go

```diff
@@ -0,0 +1,102 @@
+package hasher
+
+import (
+	"encoding/binary"
+	"fmt"
+	"hash"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+// Hasher generates a unique binary identifier for a longer piece of binary data
+// using an efficient, non-cryptographic hash function.
+type Hasher interface {
+	// CreateID is an idempotent method for creating a unique shortened identifier
+	// for the provided binary value.
+	CreateID(value []byte, collisions int) []byte
+}
+
+// NewHasher creates a new hasher instance. Default parameters are currently set to use the first
+// 4-bytes of the 64-bit BLAKE2b, non-cryptographic hash. In the case of a collision, more bytes
+// of the hash will be used for disambiguation but this happens in a minority of cases except
+// for massively large data sets.
+func NewHasher() (Hasher, error) {
+	return NewHasherWithOptions(HashOptions{})
+}
+
+// NewHasherWithOptions creates a Hash with custom options. Most users should just use NewHasher
+// with the default values.
+func NewHasherWithOptions(options HashOptions) (Hasher, error) {
+	minLength := options.MinLength
+	if minLength == 0 {
+		minLength = 4
+	}
+
+	newHash := options.NewHash
+	if newHash == nil {
+		newHash = func() hash.Hash {
+			hash, err := blake2b.New512(nil)
```

Comment:
why 512? i think this should be
```suggestion
			hash, err := blake2b.New(8, nil)
```

Reply:
Updated in https://github.com/regen-network/regen-ledger/pull/970/commits/425bb45ea5077b4ec0664c9d59cec0c62386a3c7